base/gmp.jl

```diff
@@ -174,8 +174,12 @@ end
 
 invert!(x::BigInt, a::BigInt, b::BigInt) =
     ccall((:__gmpz_invert, libgmp), Cint, (mpz_t, mpz_t, mpz_t), x, a, b)
-invert(a::BigInt, b::BigInt) = invert!(BigInt(), a, b)
 invert!(x::BigInt, b::BigInt) = invert!(x, x, b)
+function invert(a::BigInt, b::BigInt)
+    ret = deepcopy(a)
```

Comment:
```suggestion
    ret = BigInt()
```
A `deepcopy` is a bit excessive.

P.S.: Alternatively you can fix the `invert!(x::BigInt, a::BigInt, b::BigInt)` function to return `x`.

Reply:
I dont think I can, without also refactoring `invmod(x::BigInt, y::BigInt)` at [line 519](https://github.com/JuliaLang/julia/blob/dde5028a11662fca831dd676505264566aec84ab/base/gmp.jl#L519)

Reply:
However I modified the original commit in shorter notation, without deepcopy.

Reply:
> Alternatively you can fix the `invert!(x::BigInt, a::BigInt, b::BigInt)` function to return `x`.

Initially I agreed, but looking at `invmod`, I see that there still needs to be a way to get the return code and the return value is the cleanest way I can think of.